homeassistant/components/device_tracker/owntracks.py

```diff
@@ -48,6 +51,56 @@ def owntracks_location_update(topic, payload, qos):
 
         see(**kwargs)
 
-    mqtt.subscribe(hass, LOCATION_TOPIC, owntracks_location_update, 1)
+    def owntracks_event_update(topic, payload, qos):
+        """ MQTT event (geofences) received. """
+
+        # Docs on available data:
+        # http://owntracks.org/booklet/tech/json/#_typetransition
+        try:
+            data = json.loads(payload)
+        except ValueError:
+            # If invalid JSON
+            logging.getLogger(__name__).error(
+                'Unable to parse payload as JSON: %s', payload)
+            return
+
+        if not isinstance(data, dict) or data.get('_type') != 'transition':
+            return
+
+        # check if in "home" fence or other zone
+        location = ''
+        if data['event'] == 'enter':
+
+            if data['desc'] == 'home':
+                location = STATE_HOME
+            else:
+                location = data['desc']
+
+        elif data['event'] == 'leave':
+            location = STATE_NOT_HOME
```

Comment:
Thanks. I did some testing and this doesn't seem to be an issue with owntracks (at least it didn't happen in my test setup)